src/DotNetEnv/Configuration/EnvConfigurationProvider.cs

```diff
@@ -37,22 +38,13 @@ public override void Load()
             }
 
             // Since the Load method does not take care of cloberring, We have to check it here!
-            foreach (var value in values)
-            {
-                var key = NormalizeKey(value.Key);
-                if (this.options.ClobberExistingVars)
-                {
-                    this.Data[key] = value.Value;
-                }
-                else
-                {
-                    if (!this.Data.ContainsKey(key))
-                    {
-                        this.Data.Add(key, value.Value);
-                    }
-                }
-            }
+            var dotEnvDictionary = values.ToDotEnvDictionary(options.ClobberExistingVars ? CreateDictionaryOption.TakeLast : CreateDictionaryOption.TakeFirst);
```

Comment:
Oh, I meant just pull the ternary operation into its own var. I find conditionals better to have on their own, and then you could leave the ToDotEnvDictionary in the for loop.

Reply:
looks better ;)
done